decoders/netflow/netflow.go

```diff
@@ -165,9 +165,10 @@ func GetTemplateSize(template []Field) int {
 
 // DecodeDataRecordFields decodes the fields(type and value) of DataRecord.
 func DecodeDataRecordFields(payload *bytes.Buffer, listFields []Field, record *[]DataField) error {
-	*record = (*record)[:cap(*record)]
-	if len(*record) < len(listFields) {
-		*record = append(*record, make([]DataField, len(listFields)-len(*record))...)
+	if cap(*record) < len(listFields) {
+		*record = make([]DataField, len(listFields))
```

Comment:
that makes you lose previously allocated memory because you're essentially nullifying the suggested slice entirely. I'd come up with smth like this:
```
    if d := len(listFields) - cap(*record); d > 0 {
        *record = append((*record)[:cap(*record)], make([]DataField, d)...)
    } else {
        *record = (*record)[:len(listFields)]
    }
```